apps/src/javalab/JavalabButton.jsx

```diff
@@ -28,11 +29,13 @@ function JavalabButton({
     <button
       id={id}
       type="button"
-      className={className}
+      className={classNames(
+        moduleStyle.button,
+        isHorizontal && moduleStyle.buttonHorizontal,
```

Comment:
I like what you did here. How did you decide to move this into modules, but keep the disabled in css-in-js?

Reply:
Oh, I see ... the color could be passed in? I think you could either (a) concatenate the disabled styling *after* the `className` prop gets applied so that it overrides whatever custom styling, or (b) remove the `className` prop entirely––it's not being used right now (from what I can see), and there's already a `style` prop and it's weird to have both

Reply:
re: a), by "concatenate the disabled styling", do you mean add a disabled class name and add it to the list of classnames? The situation I'm aiming for is that JavalabButton can control what it looks like when it's disabled, and I can't figure out how to do that using class names alone. Maddie helped me understand that [passed in classes will always override classes defined within a specific component](https://codedotorg.slack.com/archives/C03MZ259SUU/p1660173211591569?thread_ts=1659471687.217699&cid=C03MZ259SUU), which makes me think that if a consuming component is going to pass in color, it will always override the colors defined by a disabled class in JavalabButton

re: b) I am passing in a className prop (eg, in ControlButtons)?

Reply:
For a) I *think* you could do 
```
className={classNames(
        moduleStyle.button,
        isHorizontal && moduleStyle.buttonHorizontal,
        className,
        isDisabled && moduleStyle.disabledButton
)}
```
I can't tell in Maddie's example that you linked if it takes precedence because it's a prop, or because it comes last in the classNames function. My guess is that it's because it comes last in the classNames function ...

For b) I looked in ControlButtons and don't see where a className prop is used for JavalabButton. Can you give the line number you're referring to, or a link to the line?

Reply:
> My guess is that it's because it comes last in the classNames function ...

I'm pretty sure that isn't the case, I was playing with the order a bit when I was working on this and always saw the passed in class name taking precedence. In general in CSS it doesn't matter what order you list the classes when you apply them, but the order in which they're defined, right? There's some more discussion to that effect in that thread with Maddie. I definitely find it pretty confusing that passed in class names take precedence.

Reply:
> Thoughts about removing the style prop?

That might be doable, we have a couple components still passing in style (eg, [this one](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/codeReviewGroups/UnassignedStudentsPanel.jsx#L22))

Reply:
Ah, I see. I was hoping classNames did something clever to get around that, but I suppose not.

Ah good call, I missed that –– I'm good with the tradeoffs you made!

Reply:
Just to chime in here, I _think_ based on CSS specificity rules, if the `.disabled` class is defined as a direct child of the top level class using &, i.e. `&.disabled` then it should override any external classes? Here's an example where I was testing that out - not sure if I laid it out correctly though, so let me know if this doesn't apply: https://www.sassmeister.com/gist/e9d3453384e790cc032a2aff85665563 (view HTML to see the difference).

Reply:
ooh that sounds promising! Let me try that

Reply:
So to close the loop, this worked, but not 100% sure why. I guess a rule that is defined as only applying when two classes are applied to an element (ie, this case -- an element with both the `button` and `disabled` classes) supercedes single class names passed in via props? I wonder what happens if a parent passed in two class names and had a rule of equal specificity...